tests/rptest/tests/e2e_shadow_indexing_test.py

```diff
@@ -95,11 +99,56 @@ def test_write(self):
                                   topic=self.topic,
                                   partition_idx=0,
                                   count=6)
-
         self.start_consumer()
         self.run_validation()
 
 
+class EndToEndShadowIndexingTestWithDisruptions(EndToEndShadowIndexingBase):
+    def __init__(self, test_context):
+        super().__init__(test_context,
+                         extra_rp_conf={
+                             'default_topic_replications': self.num_brokers,
+                         })
+
+    @cluster(num_nodes=5, log_allow_list=CHAOS_LOG_ALLOW_LIST)
+    def test_write_with_node_failures(self):
+        self.start_producer()
+        produce_until_segments(
+            redpanda=self.redpanda,
+            topic=self.topic,
+            partition_idx=0,
+            count=10,
+        )
+
+        self.kafka_tools.alter_topic_config(
+            self.topic,
+            {
+                TopicSpec.PROPERTY_RETENTION_BYTES:
+                5 * EndToEndShadowIndexingTest.segment_size
+            },
+        )
+
+        with random_process_kills(self.redpanda) as ctx:
+            wait_for_segments_removal(redpanda=self.redpanda,
+                                      topic=self.topic,
+                                      partition_idx=0,
+                                      count=6)
+            self.start_consumer()
+            self.run_validation()
+
+        action_log = ctx.action_log()
+        assert action_log
+
+        # At least one process kill action is present
+        assert action_log[0].is_reverse_action is False
+
+        # If there was another action after process kill, it was revival,
+        # because we are running in reverse after action mode
+        if len(action_log) > 1:
+            assert action_log[1].is_reverse_action and action_log[
+                1].node == action_log[0].node
```

Comment:
this check is excessive especially if we remove `restore_state_on_exit` and it requires to track action_log as part of `ActionInjectorThread` we may achieve the same effect if `ProcessKill` internally asserts that `.reverse` must follow `.action`

Reply:
Sure, will cover this as part of the general code change with removing restore etc.

Reply:
fixed in latest commit